spec/spec_helper_acceptance.rb

```diff
@@ -64,108 +51,55 @@ def run_device(options = { allow_changes: true })
                           else
                             [0, 2]
                           end
-  on(default, puppet('device', '--verbose', '--trace'), acceptable_exit_codes: acceptable_exit_codes) do |result|
-    # on(default, puppet('device','--verbose','--color','false','--user','root','--trace','--server',master.to_s), { :acceptable_exit_codes => acceptable_exit_codes }) do |result|
-    if options[:allow_changes] == false
-      expect(result.stdout).not_to match(%r{^Notice: /Stage\[main\]})
-    end
-    expect(result.stderr).not_to match(%r{^Error:})
-    expect(result.stderr).not_to match(%r{^Warning:})
+  result = Open3.capture2e("bundle exec puppet device --apply #{@file.path} --verbose --trace --modulepath spec/fixtures/modules --deviceconfig spec/fixtures/acceptance-device.conf --target sut")
+  if options[:allow_changes] == false
+    expect(result[0]).not_to match(%r{^Notice: /Stage\[main\]})
   end
+  expect(result[0]).not_to match %r{Error:}
+  expect(result[0]).not_to match %r{Warning:}
 end
 
 def run_resource(resource_type, resource_title = nil)
   if resource_title
-    on(master, puppet('device', '--target', 'target', '--resource', resource_type, resource_title, '--trace'), acceptable_exit_codes: [0, 1]).stdout
+    result = Open3.capture2e("bundle exec puppet device --target sut --resource #{resource_type} #{resource_title} --trace --modulepath spec/fixtures/modules --deviceconfig spec/fixtures/acceptance-device.conf")
   else
-    on(master, puppet('device', '--target', 'target', '--resource', resource_type, '--trace'), acceptable_exit_codes: [0, 1]).stdout
+    result = Open3.capture2e("bundle exec puppet device --target sut --resource #{resource_type} --trace --modulepath spec/fixtures/modules --deviceconfig spec/fixtures/acceptance-device.conf")
   end
+  result[0]
 end
 
 def device_model
-  fact('hardwaremodel')
+  fact['hardwaremodel'][%r{(\d\d\d\d)}, 1]
+end
+
+def fact
+  result = Open3.capture2e('bundle exec puppet device --target sut --modulepath spec/fixtures/modules --deviceconfig spec/fixtures/acceptance-device.conf --facts')
+  JSON.parse(result[0])['values']
 end
 
 RSpec.configure do |c|
   c.before :suite do
-    unless ENV['BEAKER_TESTMODE'] == 'local'
-      unless ENV['BEAKER_provision'] == 'no'
+    system('rake spec_prep')
+    c.add_setting :host, default: device_ip
+    c.add_setting :user, default: device_user
+    c.add_setting :password, default: (device_password || 'admin')
```

Comment:
Not sure if we want to set a default password or enable_password as forgetting to include the enable password and it resetting it to `admin` or something could cause a lot of issues. I feel, `host`, `user`, `password` and `enable_password` should all be mandatory.